web/views/student.py

```diff
@@ -1078,6 +1073,34 @@ def upload_results(request, assignment_id, submit_num, login):
     return JsonResponse({"success": True})
 
 
+def mark_solution_as_final(request, assignment_id, login, submit_num):
+    submit = get_object_or_404(
```

Comment:
This allows anyone to mark any submit as being final. Let's:
- Add `@login_required`, to ensure that only logged in users can do that
- Check that the owner of the submit (`login`) matches the username of the logged in user, otherwise raise `PermissionDenied` (see e.g. `task_detail`) for an example.
- Check that the submit being marked as final is the most recent one for the given assignment and student.